src/components/TokenSelect/TokenSelect.styles.tsx

```diff
@@ -0,0 +1,59 @@
+import styled from 'styled-components';
+import IconButton from '../IconButton/IconButton';
+import LoadingSpinner from '../LoadingSpinner/LoadingSpinner';
+import TextInput from '../TextInput/TextInput';
+
+type StyledTokenSelectProps = {
+  hasToken: boolean;
+}
+
+export const TokenSelectorLoader = styled(LoadingSpinner)`
+  position: absolute;
+  bottom: 0;
+  right: 0;
+  width: 2.5rem;
+  height: 2.5rem;
+  padding: 0.25rem;
+`;
+
+export const TokenSelectorButton = styled(IconButton)``;
+
+export const StyledTextInput = styled(TextInput)``;
+
+export const StyledTokenSelect = styled.div<StyledTokenSelectProps>`
+  position: relative;
+  width: 100%;
+  
+  & + & {
+    margin-top: 1.5rem;
+  }
+  
+  ${TokenSelectorButton} {
+    position: absolute;
+    bottom: 0;
+    right: 0;
+    height: 2.5rem;
+    padding: 0.25rem;
+    color: ${(props) => props.hasToken ? props.theme.colors.white : props.theme.colors.lightGrey};
+    
+    &:hover {
+      color: ${(props) => props.theme.colors.white};
+    }
+
+    svg {
+      margin-left: 1rem;
+      color: ${(props) => props.theme.colors.primary};
+    }
+  }
+
+  ${StyledTextInput} {
+
+    label {
+      margin-bottom: 0.125rem;
+    }
+
+    input {
+      padding-bottom: 0.5rem;
+    }
```

Comment:
Is this the way to style the children of another component?

Reply:
To me it feels weird to style the child element of another component. I think it's DOM structure should be irrelevant to other components.

I think there are three options:

1. Preset layouts for the component as above
2. Expose several classNames style props (e.g. labelClassNames, inputClassNames) which accept a css class name and apply it to the label. Not 100% sure how this works with styled components but assume something to that effect is possible.
3. (Similar to 2), expose a labelCss prop (and one for the input) which expects styled component css from the css utility (FlattenSimpleInterpolation seems to be the type). Ref here: https://github.com/airswap/airswap-web/blob/e9e6750dcec2d976a55638e5a1e806f100ea016a/src/components/Typography/Typography.styles.tsx#L59

Reply:
I found a solution, in TokenSelect.styles.tsx you can import the TextInput child elements:

```
export const StyledTokenSelect = styled.div<StyledTokenSelectProps>`

  ${StyledTextInput} {
    
    ${StyledFormLabel} {
      margin-bottom: 0.125rem;
    }
    
    ${StyledInput} {
      padding-bottom: 0.5rem;
    }    
  }
}

```